pgrx-tests/src/framework.rs

```diff
@@ -530,13 +597,27 @@ fn monitor_pg(mut command: Command, cmd_string: String, loglines: LogLines) -> S
         // exits. TODO: Consider finding a way to handle cases where we fail to
         // clean up due to a SIGNAL?
         add_shutdown_hook(move || unsafe {
-            libc::kill(pid as libc::pid_t, libc::SIGTERM);
-            let message_string = std::ffi::CString::new(
-                format!("stopping postgres (pid={pid})\n").bold().blue().to_string(),
-            )
-            .unwrap();
-            // IMPORTANT: Rust string literals are not naturally null-terminated
-            libc::printf("%s\0".as_ptr().cast(), message_string.as_ptr());
+            if let Some(_runas) = get_runas() {
+                Command::new("sudo")
+                    .arg("-u")
+                    .arg("root") // NB:  we must be "root" to kill the `sudo` process we spawned to start postgres
+                    .arg("kill")
+                    .arg("-s")
+                    .arg("SIGTERM")
+                    .arg(pid.to_string())
+                    .spawn()
+                    .expect("failed to spawn `sudo kill`")
+                    .wait()
+                    .expect("`sudo kill` didn't work");
```

Comment:
...thinking about how this could get weird. the extension we just loaded could set a SIGTERM handler, I suppose?

Reply:
We're trying to shutdown the Postgres instance we started.  Normally, the user running `cargo pgrx test` owns that process.  In the case of `--runas`, it's actually "root" that owns that process.  Hence this dance to exec `sudo kill ...`